packages/legacy/src/common/components/ProviderSelect.tsx

```diff
@@ -138,14 +153,16 @@ export const ProviderSelect: React.FunctionComponent<ProviderSelectProps> = ({
           placeholderText="Select a provider..."
           isOpen={isOpen}
           onToggle={setIsOpen}
-          selections={
-            selectedProvider?.metadata.uid ? [optionsByUid[selectedProvider.metadata.uid]] : []
-          }
+          selections={selectedOptions}
           onSelect={(_event, selection) => {
             setIsOpen(false);
-            const matchingInventoryProvider = getMatchingInventoryProvider(
-              (selection as OptionWithValue<IProviderObject>).value
-            );
+            const provider = (selection as OptionWithValue<IProviderObject>).value;
+            const matchingInventoryProvider = getMatchingInventoryProvider(provider);
+
+            if (onProviderSelect) {
```

Comment:
1. note that before the callback was called last, now it's before setting the `form.field`. In this case `onProviderSelect` is taking care of that but leaving it up to the caller is risky
2.  having both `afterChange` and `onProviderSelect` is confusing